writer_test.go

```diff
@@ -432,3 +439,64 @@ func testWriterSmallBatchBytes(t *testing.T) {
 		t.Error("bad messages in partition", msgs)
 	}
 }
+
+func testWriterIdleConnDefaultTimeout(t *testing.T) {
+	const topic = "test-writer-idleconn-1"
+	createTopic(t, topic, 1)
+	w := newTestWriter(WriterConfig{
+		Topic: topic,
+	})
+	defer w.Close()
+	ctx := context.Background()
+	msg := Message{Value: []byte("Hello World")}
+	// initial message when we establish a connection with the broker
+	err := w.WriteMessages(ctx, msg)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+	// sleep for more than the broker's max idle conn time
+	time.Sleep(2300 * time.Millisecond)
+
+	// The write should fail as we will try to send the message on a connection that the broker closed
+	err = w.WriteMessages(ctx, msg)
+	if err == nil {
+		t.Errorf("Expected broken pipe error")
+	}
+	// This write should pass as the writer will establish a new connection after the previous failure.
+	err = w.WriteMessages(ctx, msg)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+}
+
+func testWriterIdleConnSpecificTimeout(t *testing.T) {
+	const topic = "test-writer-idleconn-2"
+	createTopic(t, topic, 1)
+	w := newTestWriter(WriterConfig{
+		Topic:              topic,
+		ConnMaxIdleTimeout: 1500 * time.Millisecond,
+	})
+	defer w.Close()
+	ctx := context.Background()
+	msg := Message{Value: []byte("Hello World")}
+	// initial message when we establish a connection with the broker
+	err := w.WriteMessages(ctx, msg)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+	// sleep for more than the broker's max idle conn time
+	time.Sleep(2300 * time.Millisecond)
```

Comment:
Could we make the wait time under 1s? Slow tests are a pain for development iteration.

Reply:
https://circleci.com/gh/sagarkrkv/kafka-go/tree/addConnIdleTimeout

This test is flaky even at 2s, I don't know if it's possible to set up a reliable test, thoughts?

Reply:
I tried several combinations of timeout/sleep values, but I was unable to set up a good test that can reliably simulate the kafka broker connection close issue.